src/Microsoft.Restier.Providers.EntityFramework/Model/ModelProducer.cs

```diff
@@ -62,7 +62,19 @@ public Task<IEdmModel> GetModelAsync(ModelContext context, CancellationToken can
             var dbContext = context.GetApiService<DbContext>();
 
             var efModel = (dbContext as IObjectContextAdapter).ObjectContext.MetadataWorkspace;
-            var efEntityContainer = efModel.GetItems<EntityContainer>(DataSpace.CSpace).Single();
+            var efEntityContainers = efModel.GetItems<EntityContainer>(DataSpace.CSpace);
+            var efEntityContainer = efEntityContainers.FirstOrDefault(c => c.Name == dbContext.GetType().Name);
+            if (efEntityContainer == null)
+            {
+                if (efEntityContainers.Count > 1)
+                {
+                    var containerNames = efEntityContainers.Aggregate("", (current, next) => next.Name + ", ");
+                    throw new Exception("This project has multiple EntityFrameworkApis using different DbContexts, and the correct context could not be loaded. \r\n" +
+                        $"The contexts available are '{containerNames.Substring(0, containerNames.Length - 2)}' but the Container expects '{efEntityContainer.Name}'.");
+                }
```

Comment:
I hard-coded the string for now so that you could see the exception that is thrown, and there can be a possible discussion about the language, if necessary. I also wanted the team to see how to throw an informative exception (most of the exceptions in Restier are not very informative). We can move the strings into the property files after everyone is OIK with it.